pkg/gateway/gateway.go

```diff
@@ -339,8 +339,15 @@ func (g *gatewayType) onStoppedLeading() {
 
 	logger.Info("Controllers stopped")
 
-	if err := g.gatewayPod.SetHALabels(context.Background(), subv1.HAStatusPassive); err != nil {
-		logger.Warningf("Error updating pod label to passive: %s", err)
+	ctx := context.Background()
+	if err := wait.PollUntilContextCancel(ctx, 100*time.Millisecond, true, func(ctx context.Context) (bool, error) {
+		if g.gatewayPod.SetHALabels(ctx, subv1.HAStatusPassive) != nil {
+			return false, nil //nolint:nilerr // keep trying
+		}
+
+		return true, nil
+	}); err != nil {
+		logger.Errorf(err, "Failed updating pod label to passive: %s", err)
```

Comment:
A few things here... re: prior comment https://github.com/submariner-io/submariner/pull/3016#discussion_r1611798154, we won't get any indication of a failure to update b/c we don't return an error to `PollUntilContextCancel` so it would only return an error if the passed `ctx` is cancelled. However, since we pass the `Background` context which is never cancelled, `PollUntilContextCancel` won't return unless/until `SetHALabels` succeeds. We would expect it to eventually succeed if it's a transient error but, if not, it would silently loop forever. Also if the pod was stopped, it wouldn't gracefully exit.

I propose:

1) Use `retry.OnError` to filter out errors that aren't transient (like NotFound or permission-related) so that it returns the error and we log it 
2) Log the first non-transient error and indicate that it's retrying
3) Use the original `ctx` that was passed to `Run`, ie have `startLeaderElection` pass the `ctx` to `onStoppedLeading`

I'll submit a follow-up PR.

Reply:
I submitted https://github.com/submariner-io/submariner/pull/3018 (since you were off the rest of the day).